utils/fixed_decimal/src/decimal.rs

```diff
@@ -901,27 +901,44 @@ pub enum RoundingMode {
 
 #[cfg(feature = "ryu")]
 impl FixedDecimal {
-    /// Construct a [`FixedDecimal`] from an f64. This uses `ryu` and
-    /// goes through an intermediate string representation, so is not
-    /// fully efficient. See [icu4x#166](https://github.com/unicode-org/icu4x/issues/166) for
-    /// more details.
+    /// Construct a [`FixedDecimal`] from an f64.
+    ///
+    /// Since f64 values do not carry a notion of their precision, the second argument to this
+    /// function specifies the type of precision associated with the f64. For more information,
+    /// see [`DoublePrecision`].
+    ///
+    /// This function uses `ryu`, which is an efficient double-to-string algorithm, but other
+    /// implementations may yield higher performance; for more details, see
+    /// [icu4x#166](https://github.com/unicode-org/icu4x/issues/166).
     ///
     /// This function can be made available with the `"ryu"` feature.
     ///
     /// ```rust
-    /// use fixed_decimal::{DoublePrecision, FixedDecimal};
+    /// use fixed_decimal::{DoublePrecision, FixedDecimal, RoundingMode};
     /// use writeable::Writeable;
     ///
-    /// let decimal = FixedDecimal::new_from_f64(0.012345678, DoublePrecision::Maximum).unwrap();
+    /// let decimal = FixedDecimal::try_from_f64(
+    ///     -5.1,
+    ///     DoublePrecision::Magnitude(-2, RoundingMode::Unnecessary)
+    /// )
+    /// .expect("Finite quantity with limited precision");
+    /// assert_eq!(decimal.write_to_string(), "-5.10");
+    ///
+    /// let decimal = FixedDecimal::try_from_f64(
+    ///     0.012345678,
+    ///     DoublePrecision::Floating
+    /// )
+    /// .expect("Finite quantity");
     /// assert_eq!(decimal.write_to_string(), "0.012345678");
     ///
-    /// let decimal = FixedDecimal::new_from_f64(-123456.78, DoublePrecision::Maximum).unwrap();
-    /// assert_eq!(decimal.write_to_string(), "-123456.78");
-    ///
-    /// let decimal = FixedDecimal::new_from_f64(12345678000., DoublePrecision::Maximum).unwrap();
+    /// let decimal = FixedDecimal::try_from_f64(
+    ///     12345678000.,
```

Comment:
`nit`: strenuous dot?

Reply:
necessary for it to be a float

Reply:
could we do `12345678000f64` instead?

Reply:
I find it much harder to read, much easier to miss when reading and generally less explicit which when dealing with nuances (type of number is a nuance) is a bad practice, but if that more idiomatic, then I'll stay on the wrong side of history but not block :)